vaadin/docs
# feat: setup for docbuilder

Fixes https://github.com/vaadin/docs-app/issues/106
Fixes https://github.com/vaadin/docs-app/issues/200

This PR adds commands for developing and building the project using the [Docbuilder image](https://github.com/vaadin/docs-app/pull/211).

Prerequisites:
- Docker installed https://docs.docker.com/get-docker/
- Java, Maven, and NodeJS installed (for license checking purposes)
- Docbuilder image [installed locally](https://github.com/vaadin/docs-app/pull/211) (not yet published to Docker Hub)

Testing the PR:
1. Make sure your environment is clean `npm run docbuilder:clean`
2. Start-up the app in development mode: `npm run docbuilder:start`
   - Note that the first run will take a lot of time. Subsequent runs will be faster
   - Once the server is up and running, the app is available at [`localhost:8000`](http://localhost:8000)
3. Build the app: `npm run docbuilder:build`
   - Note that the first run will take a lot of time. Subsequent runs will be faster
   - Once the build is complete, you can find the following build artifacts
     - `/docbuilder/build/out/[docs].jar` (the docs Vaadin project production build jar)
     - `/docbuilder/build/out/public` (the static pages build by docs-app)
   - You can configure the build with environment variables listed [here](https://github.com/vaadin/docs-app#list-of-environment-variables)
     - example: `DOCS_PATH_PREFIX=/docs/latest npm run docbuilder:build`
     - alternatively, you can create an `.env` file with the `key="value"` pairs under `docbuilder/develop` and `docbuilder/build` and start with `npm run docbuilder:build`
4. The build artifacts from the previous steps can be deployed in various different ways. This PR adds one deployment example (using Docker compose)
  4.1. `cd docbuilder/build/deployment-example-docker`
  4.2. `docker-compose up`
     - Once the server is up and running, the app is available at [`localhost:80`](http://localhost:80)
